assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js

```diff
@@ -87,6 +118,49 @@ const DashboardIdeasWidget = ( {
 	);
 	const activeTab = DashboardIdeasWidget.tabIDsByIndex[ activeTabIndex ];
 
+	const [ ideaHubContainer, inView ] = useInView( {
+		triggerOnce: true,
+		threshold: 0.25,
+	} );
+
+	useEffect( () => {
+		if ( inView ) {
+			trackEvent( IDEA_HUB_GA_CATEGORY_WIDGET, 'widget_view' );
+		}
+	}, [ inView ] );
+
+	useEffect( () => {
+		// Do nothing if the following events have already been tracked
+		// or the widget hasn't appeared in the viewport yet.
+		if ( trackedWidgetView || ! inView ) {
+			return;
+		}
+
+		if ( hasNoIdeas ) {
+			trackEvent(
+				IDEA_HUB_GA_CATEGORY_WIDGET,
+				'widget_gathering_data_view'
+			);
+		} else if ( hasManyIdeas ) {
+			trackEvent(
+				IDEA_HUB_GA_CATEGORY_WIDGET,
+				'default_tab_view',
+				DashboardIdeasWidget.tabIDsByIndex[ activeTabIndex ]
+			);
+		}
+
+		if ( hasNoIdeas || hasManyIdeas ) {
+			setTrackedWidgetView( true );
+		}
```

Comment:
Let's move this call right before the `trackEvent`s above rather than duplicating the condition again here. It means duplicating the set state call once but it's a bit simpler and also makes it a little more clear what the state represents 🙂  It could go right after too, I generally prefer to set it closer to the guard and before the action (once qualified) rather than after.

Reply:
Updated.